server/index.js

```diff
@@ -51,6 +51,91 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId) {
+    return res.json([]);
+  } else {
+    const value = [req.session.cartId];
+    const text = `
+    SELECT "c"."cartItemId",
+           "c"."price",
+           "p"."productId",
+           "p"."image",
+           "p"."name",
+           "p"."shortDescription"
+      FROM "cartItems" AS "c"
+      JOIN "products" AS "p" USING ("productId")
+     WHERE "c"."cartId" = $1;
+     `;
+    db.query(text, value)
+      .then(result => res.json(result.rows[0]));
+  }
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const value = [req.body.productId];
+  const text = `
+  SELECT  "price"
+  FROM    "products"
+  WHERE   "productId" = $1;
+  `;
+  const parsedId = parseInt(req.body.productId);
+  if (parsedId < 0 || isNaN(parsedId)) {
+    return res.status(400).json({ error: 'Invalid Product ID' });
+  }
+  db.query(text, value)
+    .then(priceResult => {
+      const price = priceResult.rows[0];
+      if (!priceResult.rows[0]) {
+        next(new ClientError('Product not found', 400));
```

Comment:
Here you need to `throw` the error, because calling next does not stop the promise chain.

Reply:
Submitted suggested change